ios/Classes/AblyFlutterPlugin.m

```diff
@@ -52,11 +52,14 @@ -(void)registerWithCompletionHandler:(FlutterResult)completionHandler;
     AblyFlutterMessage *const messageData = message.message;
     NSMutableDictionary<NSString *, NSObject *> *const _dataMap = messageData.message;
     NSString *const channelName = (NSString*)[_dataMap objectForKey:@"channel"];
-    NSString *const eventName = (NSString*)[_dataMap objectForKey:@"name"];
+    NSString * eventName = (NSString*)[_dataMap objectForKey:@"name"];
+    if(eventName==[NSNull null]){
```

Comment:
@QuintinWillison any better way to do this?

Dart side will not be sending the key if there is no value, and ably-cocoa will throw errors when name is NSNull

https://github.com/ably/ably-cocoa/blob/main/Source/ARTMessage.m#L52

Reply:
I think just leave it as is for now - it's an internal detail so we can see how it scales over time.